packages/protocol/contracts/governance/Governance.sol

```diff
@@ -1252,6 +1255,43 @@ contract Governance is
     }
   }
 
+  /**
+   * @notice Removes the proposal from the queue if `lastDequeue` time has passed.
+   * @param proposalId The ID of the proposal.
+   * @dev If any of the top proposals have expired, they are deleted.
+   */
+  function dequeueProposalIfReady(uint256 proposalId) public returns (bool isProposalDequeued) {
+    isProposalDequeued = false;
+    // solhint-disable-next-line not-rely-on-time
+    if (now >= lastDequeue.add(dequeueFrequency)) {
+      Proposals.Proposal storage proposal = proposals[proposalId];
+      // Updating refunds back to proposer
+      refundedDeposits[proposal.proposer] = refundedDeposits[proposal.proposer].add(
```

Comment:
This seems like lots of duplicate code - can we merge it with code in dequeueProposalsIfReady ? Meaning that dequeueProposalsIfReady would call this function for every proposal

Reply:
Thanks for your feedback. The goal of our decision before i separated the function as such is so that we can deprecate the `dequeueProposalsIfReady` at some point. will bring this up again.